packages/utils/src/terminology-server/snomed-dedup-and-filter.ts

```diff
@@ -0,0 +1,363 @@
+        entries.splice(i, 1);
+      }
+    }
+  }
+  fs.writeFileSync(filePath, JSON.stringify(data, null, 2), "utf8");
+}
+
+function convertMedicationAdministrationsToMedicationStatements({
+  filePath,
+}: {
+  filePath: string;
+}) {
+  const data = JSON.parse(fs.readFileSync(filePath, "utf8"));
+  const entries = data.bundle ? data.bundle.entry : data.entry;
+  for (const entry of entries) {
+    const resource = entry.resource;
+    if (resource && resource.resourceType === "MedicationAdministration") {
+      entry.resource = convertMedicationAdministrationToStatement(resource);
+    }
+  }
+  fs.writeFileSync(filePath, JSON.stringify(data, null, 2), "utf8");
+}
+
+export async function fullProcessing(directoryPath: string) {
+  const removalStats = createInitialRemovalStats();
+
+  // Removal of non-disorder SNOMED codes, invalid CPT codes, and duplicate CPT codes
+  const cptSet = new Set<string>();
+  const conditionSet = new Set<string>();
+  const allRemainingEnries = new Set<string>();
+  const medicationDuplicates = new Map<string, string[]>();
+  const medicationToMedicationStatementOrAdministrationMap = new Map<
+    string,
+    { medicationStatement: string; startDate: string }
+  >();
+  await processDirectoryOrFile(directoryPath, async filePath => {
+    await removeConditionsProceduresMedAdmins({
+      filePath,
+      conditionSet,
+      cptSet,
+      allRemainingEnries,
+      removalStats,
+      medicationDuplicates,
+      medicationToMedicationStatementOrAdministrationMap,
+    });
+  });
+
+  prettyPrintRemovalStats(removalStats);
+
+  // Create hierarchy of SNOMED codes
+  const hashTable: Record<string, SnomedHierarchyTableEntry> = {};
+  await processDirectoryOrFile(directoryPath, async filePath => {
+    await computeHashTable(filePath, hashTable);
+  });
+
+  // Remove non-root SNOMED codes
+  const conditionIdsDictionary = new Set<string>();
+  await processDirectoryOrFile(directoryPath, async filePath => {
+    await removeNonRootSnomedCodes(
+      filePath,
+      hashTable,
+      allRemainingEnries,
+      conditionIdsDictionary,
+      removalStats
+    );
+  });
+
+  // Removal of duplicate RXNORM codes and medications without condition reference
+  await processDirectoryOrFile(directoryPath, async filePath => {
+    await filterMedicationStatements({
+      filePath,
+      medicationDuplicates,
+      medicationToMedicationStatementOrAdministrationMap,
+      removalStats,
+    });
+  });
+
+  await processDirectoryOrFile(directoryPath, async filePath => {
+    await convertMedicationAdministrationsToMedicationStatements({
+      filePath,
+    });
+  });
+
+  prettyPrintRemovalStats(removalStats);
+}
+
+async function main() {
+  const [directoryPath] = process.argv.slice(2);
+  if (!directoryPath) {
+    console.error("Please provide a directory path as an argument.");
+    process.exit(1);
+  }
+  await fullProcessing(directoryPath);
+}
```

Comment:
nit: It's a pain to have to scroll to the bottom and make the way up instead of just following the text from top to bottom as one scrolls it down